src/Mbstring/Mbstring.php

```diff
@@ -77,6 +77,25 @@ final class Mbstring
         array('μ', 's', 'ι',        'σ', 'β',        'θ',        'φ',        'π',        'κ',        'ρ',        'ε',        "\xE1\xB9\xA1", 'ι'),
     );
 
+    /**
+     * A helper method to either trigger an error, or throw an exception in PHP 8.
+     *
+     * @param string $error_msg  the error message to be used in the error or exception
+     * @param int    $error_type Type of the error. Not used when an exception is thrown.
+     * @param null   $exception  in case an exception must be thrown in PHP 8,
+     *                           the fully-qualified name of the exception
+     */
+    private static function triggerErrorOrException(
+      $error_msg,
+      $error_type = E_USER_NOTICE,
+      $exception = null
+    ) {
+        if (null !== $exception && PHP_VERSION_ID >= 80000) {
```

Comment:
I'd rather inline this than each place that creating this method. It will be more efficient, as OPCache can optimize the `PHP_VERSION_ID` check at compile time, and it will avoid passing class names as string.

And btw, this might also help in case the message needs to change in the PHP 8 exception.